modules/http/http-worker.c

```diff
@@ -438,6 +438,9 @@ _flush(LogThreadedDestWorker *s)
   if (self->super.batch_size == 0)
     return LTR_SUCCESS;
 
+  if (expedite)
+    return LTR_RETRY;
```

Comment:
Do you want to support the "quick-reload" only in the http() destination first?

Reply:
Is it true that this logic can not be generalized and moved to the threaded destination driver, because driver implementations might want to decide whether they want the `expedite` feature?

Reply:
Re: comment#1

I implemented "expedite" because of kafka (where we might easily have 100k messages waiting to be sent), but I implemented it in HTTP first. It would make a lot of sense to do this for SQL and anything where "flush" can be a heavy operation.

Re: comment#2

On the generalization thing, it cannot easily be generalized, as drivers tend do differ how they handle in-transit messages. For HTTP, we are building a local buffer, communication is not started to the server. For SQL, we are generating INSERT statements and a flush is basically a COMMIT. For kafka, we are similar to SQL, we post the messages that is asynchronously sent to the kafka broker, and flush is basically waiting for the thread to finish processing.

Whether it makes sense to expedite a specific driver, or how that needs to be done, cannot easily be generalized. We could probably create a "drop-queued-messages-before-reload" kind of method, that would basically be the expedite=TRUE case. But on the other hand, with kafka, in case expedite is TRUE, we are waiting 1sec for the kafka worker threads to do a final batch of messages, then we purge everything that remains. This is not a complete drop, but still ensures that the reload will be timely.

Hope this clarifies things.

Reply:
@bazsi  Thank you!